File: 01 Codigo/jsonfiles/javascript/model/Doll.js
class Doll {
    
    constructor(id, name, material, price) {
        this.id = id;
        this.name = name;
        this.material = material;
        this.price = price;
    }

    toDict() {
        return {
            id: this.id,
            name: this.name,
            material: this.material,
            price: this.price
        };
    }

    static fromDict(data) {
        return new Doll(data.id, data.name, data.material, data.price);
    }

    toString() {
        return `Doll(ID: ${this.id}, Name: ${this.name}, Material: ${this.material}, Price: ${this.price})`;
    }
}

export default Doll;

File: 01 Codigo/jsonfiles/pyhton/model/Doll.py
import json
import os
import json
import sys 

class Doll:
    def __init__(self, id, name, material, price):
        self.id = id
        self.name = name
        self.material = material
        self.price = price

    def __str__(self):
        return f"Doll(ID: {self.id}, Name: {self.name}, Material: {self.material}, Price: {self.price})"

    def to_dict(self):
        return {
            "id": self.id,
            "name": self.name,
            "material": self.material,
            "price": self.price
        }

    @staticmethod
    def from_dict(data):
        return Doll(data["id"], data["name"], data["material"], data["price"])

